### ISSUE TITLE
"Encountered an error while Applying Audio to Notes: No Voice has been set"

### ISSUE THREAD

**Issue by telotortium** [Language-Tools/anki-hyper-tts#14]
On v0.18.0. I'm using a HyperTTS key. This popup appears when I try to apply voices all the time.

This is my configuration:

```
{
    "batch_config": {
        "Chinese Grammar Wiki": {
            "source": {
                "mode": "simple",
                "source_field": "中文"
            },
            "target": {
                "remove_sound_tag": true,
                "target_field": "中文 audio",
                "text_and_sound_tag": false
            },
            "text_processing": {
                "html_to_text_line": false,
                "run_replace_rules_after": true,
                "ssml_convert_characters": false,
                "text_replacement_rules": [
                    {
                        "rule_type": "Regex",
                        "source": "^(.*?)<br/?><br/?>.*",
                        "target": "\\1"
                    }
                ]
            },
            "voice_selection": {
                "voice_list": [
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-D",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-D",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    }
                ],
                "voice_selection_mode": "random"
            }
        },
        "Example 1": {
            "source": {
                "mode": "simple",
                "source_field": "Example 1"
            },
            "target": {
                "remove_sound_tag": true,
                "target_field": "Example 1 Audio",
                "text_and_sound_tag": false
            },
            "text_processing": {
                "html_to_text_line": false,
                "run_replace_rules_after": true,
                "ssml_convert_characters": false,
                "text_replacement_rules": [
                    {
                        "rule_type": "Regex",
                        "source": "^(.*?)<br/?><br/?>.*",
                        "target": "\\1"
                    }
                ]
            },
            "voice_selection": {
                "voice_list": [
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    }
                ],
                "voice_selection_mode": "random"
            }
        },
        "Example 2": {
            "source": {
                "mode": "simple",
                "source_field": "Example 2"
            },
            "target": {
                "remove_sound_tag": true,
                "target_field": "Example 2 Audio",
                "text_and_sound_tag": false
            },
            "text_processing": {
                "html_to_text_line": false,
                "run_replace_rules_after": true,
                "ssml_convert_characters": false,
                "text_replacement_rules": [
                    {
                        "rule_type": "Regex",
                        "source": "^(.*?)<br/?><br/?>.*",
                        "target": "\\1"
                    }
                ]
            },
            "voice_selection": {
                "voice_list": [
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    }
                ],
                "voice_selection_mode": "random"
            }
        },
        "Example 3": {
            "source": {
                "mode": "simple",
                "source_field": "Example 3"
            },
            "target": {
                "remove_sound_tag": true,
                "target_field": "Example 3 Audio",
                "text_and_sound_tag": false
            },
            "text_processing": {
                "html_to_text_line": false,
                "run_replace_rules_after": true,
                "ssml_convert_characters": false,
                "text_replacement_rules": [
                    {
                        "rule_type": "Regex",
                        "source": "^(.*?)<br/?><br/?>.*",
                        "target": "\\1"
                    }
                ]
            },
            "voice_selection": {
                "voice_list": [
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-B",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-B",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Male",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-C",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-C",
                                "ssml_gender": "MALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_CN",
                            "name": "cmn-CN-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-CN",
                                "name": "cmn-CN-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    },
                    {
                        "options": {},
                        "voice": {
                            "gender": "Female",
                            "language": "zh_TW",
                            "name": "cmn-TW-Wavenet-A",
                            "service": "Google",
                            "voice_key": {
                                "language_code": "cmn-TW",
                                "name": "cmn-TW-Wavenet-A",
                                "ssml_gender": "FEMALE"
                            }
                        },
                        "weight": 1
                    }
                ],
                "voice_selection_mode": "random"
            }
        }
    },
    "configuration": {
        "hypertts_pro_api_key": "[REDACTED]",
        "service_config": {
            "Google": {
                "api_key": "[REDACTED]",
                "explorer_api_key": true
            }
        },
        "service_enabled": {
            "Google": true
        }
    }
}
```

**Comment by telotortium** [comment]
OK, I am able to get it to work, but I have to do the following:

1. Open HyperTTS menu, click on "Add Audio: My Preset"
2. Click "Apply to Notes".
3. See the error popup.
4. Click "Load" by the Preset dropdown.
5. Click "Apply to Notes".
6. Now it works.

**Comment by luc-languagetools** [comment]
Thanks for reporting, which preset are you using when this happens?

**Comment by luc-languagetools** [comment]
Got it, I will try to reproduce the issue.

**Comment by luc-languagetools** [comment]
How to reproduce:

1. from browser menu, create a preset with "random" voice selection mode and add two voices
2. save the preset
3. close browser, reopen
4. HyperTTS menu, Add Audio: My Preset
5. Click Apply to Notes.

**Comment by luc-languagetools** [comment]
I believe this is now fixed in v0.19, many thanks for reporting!

**Comment by telotortium** [comment]
Yes, it's fixed. Thanks!
